WorldMap enhancements

markdwags opened the pull request:
WorldMap reads CSV, UOAM and Ultima Mapper marker files from `Data\Client` -- *.map files and *.xml files -- and displays them on the map.

Updated context menu to show/hide specific markers
Markers only render if they are in the the viewport
Labels won't render when zoomed out at a certain level

markdwags committed:
Option to show current coordinates on WorldMap

markdwags committed:
Option to show current coordinates on WorldMap

markdwags committed:
Added option to show/hide mobiles on WorldMap

markdwags committed:
Added option to show/hide mobiles on WorldMap

markdwags committed:
Move logic to DrawAll

markdwags committed:
Move logic to DrawAll

markdwags committed:
option to show/hide your name/bar on the worldmap

markdwags committed:
option to show/hide your name/bar on the worldmap

andreakarasho committed:
+ don't allocate memory every time cuo switches between login and game screen

markdwags committed:
Removed use of StringBuilder, storing player position instead

markdwags committed:
Removed use of StringBuilder, storing player position instead

andreakarasho committed:
+ improved pooling

andreakarasho committed:
+ force label update in worldmapgump

Alias-Alan committed:
HealthBarGump+

ClassicUO - LastTarget Highlight (Red Border)
https://i.imgur.com/GSgo6QD.mp4

andreakarasho committed:
+ avoid foreach allocation for deque

andreakarasho committed:
+ use debug

roxya committed:
Tithing points added to infobar

markdwags committed:
First pass at WorldMap reading csv or UOAM marker files

markdwags committed:
Markers will only display if they are in the worldmap viewport

markdwags committed:
Option to hide/show markers

markdwags committed:
Marker labels wont show at certain zoom point

andreakarasho committed:
+ support for gump refreshing

markdwags committed:
You can define marker color

andreakarasho committed:
+ don't dispose resizepic when textures are null

markdwags committed:
Option to hide all markers or specific sets

andreakarasho committed:
+ fix bad op precedence

markdwags committed:
WorldMap reads Ultima Mapper xml files

andreakarasho commented on src/Game/UI/Gumps/WorldMapGump.cs:
```diff
@@ -86,6 +121,13 @@ public override void Restore(XmlElement xml)
             if (int.TryParse(xml.GetAttribute("zoomindex"), out int value))
                 _zoomIndex = (value >= 0 && value < _zooms.Length) ? value : 4;
 
+            _showCoordinates = bool.Parse(xml.GetAttribute("showcoordinates"));
```
bool.TryParse instead of bool.Parse 
or it will throw errors

markdwags replied:
Ok, bool.Parse is used throughout the class -- you want all of them refactored?

andreakarasho replied:
Yes just to avoid exceptions with retro compatibility

markdwags replied:
Made those changes to the bool parsing.

Alias-Alan committed:
safeguard +redundancy 

check if mobile is null before checking is lasttarget to work if mobile == null

Alias-Alan committed:
Update HealthBarGump.cs

andreakarasho committed:
gargoyle sitting

markdwags committed:
Markers can now be assigned map index & updated bool parsing

andreakarasho committed:
+ sanity check

andreakarasho committed:
improved uop data reading  and avoiding heap allocations

andreakarasho committed:
debug stuff

andreakarasho committed:
+ gargoyle equipment fix

andreakarasho committed:
old

andreakarasho committed:
+ fix multiple fly button

andreakarasho committed:
+ improved multi file reading reading
+ improved compressed multi packet reader

andreakarasho committed:
+ fix multi issue

andreakarasho committed:
+ check sitting and gargoyle special case

andreakarasho committed:
space kek

markdwags committed:
Merge branch 'dev' into map-updates

markdwags committed:
Merge branch 'dev' into map-updates
